src/main/java/uk/gov/hmcts/reform/civil/model/citizenui/CcdDashboardClaimantClaimMatcher.java

```diff
@@ -409,50 +336,107 @@ public boolean isHwFClaimSubmit() {
 
     @Override
     public boolean isHwFMoreInformationNeeded() {
-        return caseData.isHWFOutcomeReady() && caseData.getHwFEvent() == CaseEvent.MORE_INFORMATION_HWF;
+        Optional<LocalDateTime> eventTime = getTimeOfMostRecentEventOfType(EnumSet.of(CaseEvent.MORE_INFORMATION_HWF));
```

Comment:
The next 6 - 7 methods all do a slight variation of the same thing. This could be simplified and a lot of duplicated code removed.

Reply:
Should be resolved now